### PR TITLE
Added pool upgrade how-to document

### PR DESCRIPTION
**Pull request by spivachuk**
- Added pool upgrade how-to document.
- Enhanced PROPOSE_UPGRADE transaction documentation.
- Made a minor correction in pool upgrade documentation.

### CODE REVIEW

File: docs/pool-upgrade-how-to.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# Pool Upgrade How To
+
+Below there are steps which need to be done to upgrade a pool to a new DCL
+application version:
+
+1. A new release of
```
Comment (ashcherbakov) [inline docs/pool-upgrade-how-to.md:6]:
```suggestion
1. **[Dev Team] New Release and Upgrade Name**: A new release of
```

File: docs/pool-upgrade-how-to.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+   [distributed-compliance-ledger](https://github.com/zigbee-alliance/distributed-compliance-ledger)
+   project is issued. The code of the new released application version must be
+   augmented by a new upgrade handler which will handle an upgrade to this new
+   version. This upgrade handler must have a unique name which will serve as the
+   name of the upgrade to this new version.
+2. A ledger height not reached yet at which all the nodes in the pool must be
```
Comment (ashcherbakov) [inline docs/pool-upgrade-how-to.md:12]:
```suggestion
2. [Trustees] **Upgrade Height**: A ledger height not reached yet at which all the nodes in the pool must be
```

File: docs/pool-upgrade-how-to.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+   upgraded to the new application version is chosen.
+3. SHA-256 or SHA-512 checksums of the new application version binaries for
```
Comment (ashcherbakov) [inline docs/pool-upgrade-how-to.md:14]:
```suggestion
3. [A Trustee] **Release binary checksum**: SHA-256 or SHA-512 checksums of the new application version binaries for
```

File: docs/pool-upgrade-how-to.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+   supported platforms are calculated.
+4. One of trustees sends [`ProposeUpgrade`](./transactions.md#propose_upgrade)
```
Comment (ashcherbakov) [inline docs/pool-upgrade-how-to.md:16]:
```suggestion
4. [A Trustee] **ProposeUpgrade**: One of trustees sends [`ProposeUpgrade`](./transactions.md#propose_upgrade)
```

File: docs/pool-upgrade-how-to.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+   transaction with the name of the new upgrade handler, the chosen ledger
+   height and the info containing URLs of the application binaries for supported
+   platforms with the calculated checksums. So the upgrade plan is proposed.
+5. Other trustees send [`ApproveUpgrade`](./transactions.md#approve_upgrade)
```
Comment (ashcherbakov) [inline docs/pool-upgrade-how-to.md:20]:
```suggestion
5. [Trustees] **ApproveUpgrade**: Other trustees send [`ApproveUpgrade`](./transactions.md#approve_upgrade)
```

File: docs/pool-upgrade-how-to.md [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+1. A new release of
+   [distributed-compliance-ledger](https://github.com/zigbee-alliance/distributed-compliance-ledger)
+   project is issued. The code of the new released application version must be
+   augmented by a new upgrade handler which will handle an upgrade to this new
+   version. This upgrade handler must have a unique name which will serve as the
+   name of the upgrade to this new version.
+2. A ledger height not reached yet at which all the nodes in the pool must be
+   upgraded to the new application version is chosen.
+3. SHA-256 or SHA-512 checksums of the new application version binaries for
+   supported platforms are calculated.
```
Comment (ashcherbakov) [inline docs/pool-upgrade-how-to.md:15]: Please add a command example (For Ubuntu) that can be used to calculate the checksum

File: docs/pool-upgrade-how-to.md [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+4. One of trustees sends [`ProposeUpgrade`](./transactions.md#propose_upgrade)
+   transaction with the name of the new upgrade handler, the chosen ledger
+   height and the info containing URLs of the application binaries for supported
```
Comment (ashcherbakov) [inline docs/pool-upgrade-how-to.md:18]: Please add an example of the `ProposeUpgrade` command to send.

File: docs/pool-upgrade-how-to.md [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+   platforms with the calculated checksums. So the upgrade plan is proposed.
+5. Other trustees send [`ApproveUpgrade`](./transactions.md#approve_upgrade)
+   transactions with the name of the proposed upgrade until the count of
```
Comment (ashcherbakov) [inline docs/pool-upgrade-how-to.md:21]: Please add an example of the `ApproveUpgrade` command to send.

File: docs/pool-upgrade-how-to.md [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+   approvals (including the proposal) reaches 2/3 of the total count of
+   trustees. So the upgrade plan is approved and scheduled.
+6. Before the ledger reaches the height specified in the upgrade plan, each node
```
Comment (ashcherbakov) [inline docs/pool-upgrade-how-to.md:24]:
```suggestion
6. [All Node Admins] **Download new binary**: Before the ledger reaches the height specified in the upgrade plan, each node
```

File: docs/pool-upgrade-how-to.md [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+   admin does the following steps:
+    1. Downloads the application binary at the URL specified in the upgrade plan
+       `Info` field and corresponding to the node platform.
+    2. Verifies that the downloaded application binary matches the checksum
+       specified in the URL. This can be done automatically by
+       [`go-getter`](https://github.com/hashicorp/go-getter) download tool.
```
Comment (ashcherbakov) [inline docs/pool-upgrade-how-to.md:30]: Please add an example how to use go-getter for that

File: docs/pool-upgrade-how-to.md [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+5. Other trustees send [`ApproveUpgrade`](./transactions.md#approve_upgrade)
+   transactions with the name of the proposed upgrade until the count of
+   approvals (including the proposal) reaches 2/3 of the total count of
+   trustees. So the upgrade plan is approved and scheduled.
+6. Before the ledger reaches the height specified in the upgrade plan, each node
+   admin does the following steps:
+    1. Downloads the application binary at the URL specified in the upgrade plan
+       `Info` field and corresponding to the node platform.
+    2. Verifies that the downloaded application binary matches the checksum
+       specified in the URL. This can be done automatically by
```
Comment (ashcherbakov) [inline docs/pool-upgrade-how-to.md:29]: Please add an example command how to calculate the checksum (for Ubuntu)

File: docs/pool-upgrade-how-to.md [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+       [`go-getter`](https://github.com/hashicorp/go-getter) download tool.
+    3. Creates a directory with the name of the upgrade within
+       `/var/lib/<USERNAME>/.dcl/cosmovisor/` (where <USERNAME> is the name of
```
Comment (ashcherbakov) [inline docs/pool-upgrade-how-to.md:32]: Please add an example command how to create that directories

File: docs/pool-upgrade-how-to.md (hunk 6 above)
Comment (spivachuk) [reply docs/pool-upgrade-how-to.md]: Added.

File: docs/pool-upgrade-how-to.md (hunk 8 above)
Comment (spivachuk) [reply docs/pool-upgrade-how-to.md]: Added.

File: docs/pool-upgrade-how-to.md (hunk 10 above)
Comment (spivachuk) [reply docs/pool-upgrade-how-to.md]: Added.

**Review by ashcherbakov — APPROVED** [review]

File: docs/pool-upgrade-how-to.md (hunk 11 above)
Comment (spivachuk) [reply docs/pool-upgrade-how-to.md]: Added.

File: docs/pool-upgrade-how-to.md (hunk 12 above)
Comment (spivachuk) [reply docs/pool-upgrade-how-to.md:104]: Added.

**Review by ashcherbakov — APPROVED** [review]

**Pull request merged**